### PR TITLE
ManagedIdentity alignment for API Management

### PR DESCRIPTION
**Pull request by MariusStorhaug**
# Change

Restructure MI on API Management in line with what was done on AutomationAccount.

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: .github/workflows/ms.apimanagement.service.yml [hunk 1]
```diff
@@ -22,8 +22,8 @@ on:
         required: false
         default: '0.0.1'
   push:
-    branches:
-      - main
+    #branches:
```
Comment (eriqua) [inline .github/workflows/ms.apimanagement.service.yml:25]: Is commenting this trigger intentional?

Comment (MrMCake) [reply .github/workflows/ms.apimanagement.service.yml:25]: @MariusStorhaug  check it out ;)

Comment (MariusStorhaug) [reply .github/workflows/ms.apimanagement.service.yml]: Fixed

**Review by MrMCake — APPROVED** [review]

File: arm/Microsoft.ApiManagement/service/deploy.bicep [hunk 2]
```diff
@@ -425,3 +435,6 @@ output serviceResourceId string = apiManagementService.id
 
 @description('The resource group the API management service was deployed into')
 output serviceResourceGroup string = resourceGroup().name
+
+@description('The principal ID of the system assigned identity.')
+output principalId string = systemAssignedIdentity ? apiManagementService.identity.principalId : ''
```
Comment (eriqua) [inline arm/Microsoft.ApiManagement/service/deploy.bicep:440]: I'm not against this name but I see the storage account uses another `assignedIdentityID`. I'd suggest to use the same. We'll anyway need to align output names at some point (but not for this release)

Comment (MariusStorhaug) [reply arm/Microsoft.ApiManagement/service/deploy.bicep:440]: Will align all that use MI to this approach

**Pull request merged**